casbin/enforcer.py

```diff
@@ -174,19 +174,19 @@ def get_implicit_users_for_permission(self, *permission):
         get_implicit_users_for_permission("data1", "read") will get: ["alice", "bob"].
         Note: only users will be returned, roles (2nd arg in "g") will be excluded.
         """
-        subjects = self.get_all_subjects()
-        roles = self.get_all_roles()
-
-        users = set_subtract(subjects, roles)
+        p_subjects = self.get_all_subjects()
+        g_inherit = self.model.get_values_for_field_in_policy("g", "g", 1)
+        g_subjects = self.model.get_values_for_field_in_policy("g", "g", 0)
+        subjects = array_remove_duplicates(p_subjects + g_subjects)
 
         res = list()
-        for user in users:
+        for user in subjects:
             req = join_slice(user, *permission)
             allowed = self.enforce(*req)
 
             if allowed:
                 res.append(user)
-
+        res = set_subtract(res, g_inherit)
         return res
```

Comment:
Also add the test from Go: https://github.com/casbin/casbin/pull/443/files

Reply:
I believe the test for this function already exists?
![image](https://user-images.githubusercontent.com/46557895/122229229-26e35080-ceeb-11eb-94dd-6d29b32aa936.png)

Reply:
Does it already be there? How can it even work before this PR?

Reply:
@ffyuanda This test case is not comprehensive enough , like this:
```
	e.ClearPolicy()
	e.AddPolicy("admin", "data1", "read")
	e.AddPolicy("bob", "data1", "read")
	e.AddGroupingPolicy("alice", "admin")
	testGetImplicitUsers(t, e, []string{"alice", "bob"}, "data1", "read")
```

Reply:
Yeah the original test cases are not fully covered, I added the go version tests upon them.